core/src/main/java/org/sonarsource/sonarlint/core/container/global/MetadataLoader.java

```diff
@@ -28,20 +28,32 @@
 
 public class MetadataLoader {
 
-  private static final String VERSION_FILE_PATH = "/sonar-api-version.txt";
+  private static final String SONAR_PLUGIN_API_VERSION_FILE_PATH = "/sonar-api-version.txt";
+  private static final String SONARLINT_PLUGIN_API_VERSION_FILE_PATH = "/sonarlint-api-version.txt";
 
   private MetadataLoader() {
     // only static methods
   }
 
-  public static Version loadVersion(System2 system) {
-    URL url = system.getResource(VERSION_FILE_PATH);
+  public static Version loadSonarPluginApiVersion() {
+    URL url = System2.INSTANCE.getResource(SONAR_PLUGIN_API_VERSION_FILE_PATH);
```

Comment:
Why this change? Generally we prefer to inject and use `System2` - this is supposed to make the code more testable since it relies on a "mockable" instance instead of static stuff (and yes, in production code it is expected to use `System2.INSTANCE` when not in a container).

Reply:
In general I tend to agree, here we didn't have any test and the argument was always `System2.INSTANCE`, that's why I inlined it.